### ISSUE TITLE
Crash in Crisp Messaging When User Goes Offline and Clicks Push Notifications

### ISSUE THREAD

**Issue by mohsen0900** [crisp-im/crisp-sdk-android#184]
Issue Summary: We have identified a crash in the Android app that occurs when users interact with Crisp messaging under specific conditions. The crash is triggered when a user:
1. Receives a push notification related to Crisp messaging and clicks on it.
2. Sees the caption in Crisp chat: “There are new messages below. Click to see.”
3. Ignores the caption and closes the Crisp chat.
4. Goes offline from the Crisp web.
5. Receives another message from Crisp web, which triggers a new push notification.
6. Clicks on the new push notification.

Stack Trace:
```java
Fatal Exception: java.lang.IllegalArgumentException: Invalid target position
       at androidx.recyclerview.widget.RecyclerView$SmoothScroller.start(RecyclerView.java:12204)
       at androidx.recyclerview.widget.RecyclerView$LayoutManager.startSmoothScroll(RecyclerView.java:8771)
       at androidx.recyclerview.widget.LinearLayoutManager.smoothScrollToPosition(LinearLayoutManager.java:508)
       at androidx.recyclerview.widget.RecyclerView.smoothScrollToPosition(RecyclerView.java:1858)
       at im.crisp.client.internal.v.p.a(SourceFile:14)
       at im.crisp.client.internal.v.d.e(SourceFile:2)
       at im.crisp.client.internal.v.d.b(SourceFile:1)
       at im.crisp.client.internal.v.d$b.onLayoutChange(SourceFile:1)
       at android.view.View.layout(View.java:24830)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at androidx.coordinatorlayout.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1213)
       at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:899)
       at com.google.android.material.bottomsheet.BottomSheetBehavior.onLayoutChild(BottomSheetBehavior.java:576)
       at im.crisp.client.internal.utils.BottomSheetNestedScrollBehavior.onLayoutChild()
       at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:918)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
       at com.android.internal.policy.DecorView.onLayout(DecorView.java:789)
       at android.view.View.layout(View.java:24804)
       at android.view.ViewGroup.layout(ViewGroup.java:6641)
       at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:4523)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3797)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2642)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9826)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1399)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1408)
       at android.view.Choreographer.doCallbacks(Choreographer.java:1012)
       at android.view.Choreographer.doFrame(Choreographer.java:942)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1382)
       at android.os.Handler.handleCallback(Handler.java:958)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:205)
       at android.os.Looper.loop(Looper.java:294)
       at android.app.ActivityThread.main(ActivityThread.java:8394)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:640)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:995)
```

**Comment by Doc1faux** [comment]
Hi @mohsen0900 and thank you for your feedback,

I'm not sure to understand how you reproduce the issue.
The stacktrace seems to be related to Crisp SDK as it contains obfuscated lines like `im.crisp.client.internal.*` but you are talking about clicking on notifications which are fired only by the app for website conversations and not for the chatbox where you can talk with Crisp support.

Could you be more specific about where clicks on notifications lead you and on which screens you are navigating?
Could you also precise which version of the app you're using (visible at the bottom of the Crisp Settings)?

**Comment by mohsen0900** [comment]
Could you be more specific about where clicks on notifications lead you and on which screens you are navigating?
Our app has a feature where a push notification directs you to open Crisp messaging. I noticed an issue where, if new messages in the chatbox are not visible in the viewport, a prompt appears saying, "Down arrow: There are new messages below. Click to see." If I exit the chatbox without clicking on this prompt and then reopen Crisp messaging, the app crashes.

Could you also precise which version of the app you're using (visible at the bottom of the Crisp Settings)?
Crisp Console version 2024.8.1, 16:24 (4d4a7f2)
Android App version
```groovy
implementation 'im.crisp:crisp-sdk:2.0.1beta2'
```

**Comment by Doc1faux** [comment]
OK, so if I understand well, it's definitely an issue with the Android SDK integrated in your own app?
I did not get if you were talking about Crisp app or Crisp SDK ^^"

Is your notification pending intent opens an activity of your app, then this activity handles the Crisp chatbox opening?
I'll try to reproduce your crash on my side.

**Comment by mohsen0900** [comment]
It was Crisp Android SDK.

Is your notification pending intent opens an activity of your app, then this activity handles the Crisp chatbox opening?
yes.

**Comment by Doc1faux** [comment]
I was not able to reproduce it but I understood the cause with the stacktrace you provided and it should be fixed in the next `2.0.2beta3` release I should ship soon.

**Comment by Doc1faux** [comment]
I just shipped the new [`2.0.2beta3`](https://github.com/crisp-im/crisp-sdk-android/releases/tag/v2.0.2beta3) release which should fix this issue. 
It can take a couple of hours to be available.
I close the issue.
